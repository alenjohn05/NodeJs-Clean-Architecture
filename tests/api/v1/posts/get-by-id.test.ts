import { client, seed } from '../..';

describe('GET /api/v1/posts/:id', () => {
  describe('given the post does not exist', () => {
    test('responds with a 404 status code', async () => {
      const postId = Number.MAX_SAFE_INTEGER;

      const response = await client.get(`/api/v1/posts/${postId}`);

      expect(response.status).toEqual(404);
    });
  });

  describe('given a valid request', () => {
    test('responds with a 200 status code', async () => {
      const postId = seed.posts[0].id;

      const response = await client.get(`/api/v1/posts/${postId}`);

      expect(response.status).toEqual(200);
      expect(response.body).toMatchSnapshot();
    });
  });
});
