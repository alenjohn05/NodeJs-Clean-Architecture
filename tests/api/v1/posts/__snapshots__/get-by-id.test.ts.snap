// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/v1/posts/:id given a valid request responds with a 200 status code 1`] = `
{
  "id": 1,
  "published": false,
  "title": "Mock post one",
}
`;
